0.2.0/docker-entrypoint.sh

```diff
@@ -0,0 +1,279 @@
+    done
+
+}
+
+write_cql_props() {
+
+   for prop in \
+      atomic-batch-mutate \
+      batch-statement-size \
+      cluster-name \
+      compaction-strategy-class \
+      compaction-strategy-options \
+      compression \
+      compression-block-size \
+      compression-type \
+      keyspace \
+      local-datacenter \
+      only-use-local-consistency-for-system-operations \
+      protocol-version \
+      read-consistency-level \
+      replication-factor \
+      replication-strategy-class \
+      replication-strategy-options \
+      write-consistency-level \
+    ; do
+
+      write_prop ${prop} 'storage.cql.${prop}' 'JANUS_STORAGE_CQL_${prop}'
+    done
+
+    write_prop "ignored" 'storage.cql.ssl.enabled' 'JANUS_STORAGE_CQL_SSL_ENABLED'
+    write_prop "ignored" 'storage.cql.ssl.truststore.location' 'JANUS_STORAGE_CQL_SSL_TRUSTORE_LOCATION'
+    write_prop "ignored" 'storage.cql.ssl.truststore.password' 'JANUS_STORAGE_CQL_SSL_TRUSTORE_PASSWORD'
+
+}
+
+write_hbase_props() {
+
+   for prop in \
+      compat-class \
+      compression-algorithm \
+      region-count \
+      regions-per-server \
+      short-cf-names \
+      skip-schema-check \
+      table \
+    ; do
+
+      write_prop ${prop} 'storage.hbase.${prop}' 'JANUS_STORAGE_HBASE_${prop}'
+    done
+
+}
+
+write_storage_props() {
+
+  for prop in \
+      backend \
+      batch-loading \
+      buffer-size \
+      connection-timeout \
+      directory \
+      hostname \
+      page-size \
+      parallel-backend-ops \
+      password \
+      port \
+      read-only \
+      read-time \
+      setup-wait \
+      transactions \
+      username \
+      write-time \
+    ; do
+      write_prop ${prop} 'storage.${prop}' 'JANUS_STORAGE_${prop}'
+    done
+    
+    case "${JANUS_STORAGE_BACKEND}" in
+    "berkeleyje")
+        write_berkeley_props
+       ;;
+    "cassandra")
+        write_cassandra_props
+       ;;
+    "cql")
+       write_cql_props
+       ;;
+    "hbase")
+       write_hbase_props
+       ;;
+    esac
+
+}
+
+write_es_index_props() {
+
+  local index_name=${1}
+
+  for prop in \
+      client-only \
+      cluster-name \
+      health-request-timeout \
+      ignore-cluster-name \
+      interface \
+      load-default-node-settings \
+      local-mode \
+      sniff \
+      ttl-interval \
+    ; do
+
+      write_prop ${prop} 'index.'"${index_name}"'.elasticsearch.${prop}' 'JANUS_INDEX_'"${index_name^^}"'_ES_${prop}'
+  done
+
+}
+
+write_index_props() {
+
+  local index_name=${1}
+
+  for prop in \
+      backend \
+      conf-file \
+      directory \
+      hostname \
+      index-name \
+      map-name \
+      max-result-set-size \
+      port \
+    ; do
+      write_prop ${prop} 'index.'"${index_name}"'.${prop}' 'JANUS_INDEX_'"${index_name^^}"'_${prop}'
+  done
+
+  backend="JANUS_INDEX_${index_name^^}_BACKEND"
+  case "${!backend}" in
+  "elasticsearch")
+      write_es_index_props ${index_name}
+     ;;
+  esac
+
+}
```

Comment:
You might be able to avoid enumerating specific storage/index configuration items above by iterating over the environment to automatically detect/apply overrides. Modifying your write_prop function and borrowing from Elasticsearch's (Apache licensed) docker-entrypoint:

```bash
write_prop() {
  local key=${1}
  local val=${2}

  # if the line exists in the file replace it; otherwise append it
  if grep -q -E "^${key}\s*=\.*" ${JANUS_PROPS}; then
    sed -ri "s/^(${key}=).*/\\1${val}/" ${JANUS_PROPS}
  else
    echo "${key}=${val}" >> ${JANUS_PROPS}
  fi
}


write_index_and_storage_props() {
  # reference https://github.com/elastic/elasticsearch-docker/blob/821215074e4c9c3904f3eec182d8409b7cb66d7b/build/elasticsearch/bin/docker-entrypoint.sh#L51-L60  while IFS='=' read -r envvar_key envvar_value
  do
    if [[ "$envvar_key" =~ ^[a-z0-9_]+\.[a-z0-9_]+ ]]; then
      if [[ ! -z $envvar_value ]]; then
        write_prop $envvar_key $envvar_value
      fi
    fi
  done < <(env)
}
```

Reply:
This is a good idea. I started down the path to auto generate all this prop stuff so we don't have to do a manual diff on version updates but I think this may be better. Making this update locally, I'll try it out and it it works well, I'll push it on my next update.